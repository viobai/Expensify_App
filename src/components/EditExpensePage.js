import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import Modal from 'react-modal';

export class EditExpensePage extends React.Component {
    state = { show: false };

    showModal = () => {
        this.setState({ show: true });
    };

    hideModal = () => {
        this.setState({ show: false });
    };

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    

    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.setState({ show: false });
        this.props.history.push('/');
        
    }

    render() {
        return (
            <div className="page">
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.showModal}>Remove Expense</button>
                </div>
                <Modal
                    isOpen={this.state.show}
                    className="modal"
                    onRequestClose={this.hideModal}
                >
                    <div className="modal-main">
                        <h2 className="modal-title">Delete this expense?</h2>
                        <div className="modal-buttons">
                            <button className="button button--link-sm" onClick={this.hideModal}>Cancel</button>
                            <button className="button" onClick={this.onRemove}>Confirm</button>
                        </div>
                            
                    </div>
                    </Modal>
            </div>
        );
    }
}


const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);