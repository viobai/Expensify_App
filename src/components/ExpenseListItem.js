import React from 'react'; // for using jsx
import { Link } from 'react-router-dom';
//import { connect } from 'react-redux';


const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={'/edit/'+id} >
            <h3>{description}</h3>
        </Link>
        <p>{amount/100} - {createdAt}</p>
        
    </div>
);

export default (ExpenseListItem);