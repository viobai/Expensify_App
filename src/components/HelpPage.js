import React from 'react';

const HelpPage = () => (
    <div>
        This is from help.
    </div>
);

export default HelpPage;