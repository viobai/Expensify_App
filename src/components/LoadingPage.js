import React from 'react';

const LoadingPage = () => (
    <div className="loader">
        <img className="loader__image" src="/image/loader.gif" />
    </div>
);

export default LoadingPage;