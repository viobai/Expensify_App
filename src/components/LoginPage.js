import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';
//import { NavLink } from 'react-router-dom';

export const LoginPage = ({ startLogin }) => (
    <div>
        <button onClick={startLogin}>Login</button>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);