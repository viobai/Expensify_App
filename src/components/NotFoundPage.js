import React from 'react';
import { Link } from 'react-router-dom';

// <Link> swap info instead of whole page refresh, good for client side routing
const NotFoundPage = () => (
    <div>
        404 - <Link to="/">Go Home</Link>
    </div>
);

export default NotFoundPage;