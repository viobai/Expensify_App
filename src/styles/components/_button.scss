.button {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link{
    background: none;
}

.button--link-sm {
    background: none;
    color: $grey;
}

.button--secondary{
    @extend .button;
    background: #888888;
}