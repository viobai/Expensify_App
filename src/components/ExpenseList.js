import React from 'react';
import { connect } from 'react-redux'; // connect react component to redux store
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

// props defined in mapStateToProps
export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
        <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <div className="list-item list-item--message">
                        <p>No expenses</p>
                    </div>
                ) : (
                    props.expenses.map((expense) => {
                        return <ExpenseListItem key={expense.id} {...expense} />;
                    })
                )
            }
        </div>
    </div>
);

// state defined by store from <Provider>
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

// first (): define the things we want to get from store
// second (): the component that we want the version of
export default connect(mapStateToProps)(ExpenseList);