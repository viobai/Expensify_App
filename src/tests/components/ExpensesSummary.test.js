import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary with 1 expense', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
	expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with 1+ expense', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={234453} />);
	expect(wrapper).toMatchSnapshot();
});