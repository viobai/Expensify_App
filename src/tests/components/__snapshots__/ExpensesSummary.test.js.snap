// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={1}
    expensesTotal={235}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content-container"
>
        <h1
                className="page-header__title"
        >
                Viewing 
                <span>
                        1
                </span>
                 
                expense
                 totalling 
                <span>
                        $2.35
                </span>
                .
        </h1>
        <div
                className="page-header__actions"
        >
                <Link
                        className="button"
                        replace={false}
                        to="/create"
                >
                        Add Expense
                </Link>
        </div>
</div>,
      "className": "page-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-header__title"
>
            Viewing 
            <span>
                        1
            </span>
             
            expense
             totalling 
            <span>
                        $2.35
            </span>
            .
</h1>,
          <div
            className="page-header__actions"
>
            <Link
                        className="button"
                        replace={false}
                        to="/create"
            >
                        Add Expense
            </Link>
</div>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Viewing ",
              <span>
                1
</span>,
              " ",
              "expense",
              " totalling ",
              <span>
                $2.35
</span>,
              ".",
            ],
            "className": "page-header__title",
          },
          "ref": null,
          "rendered": Array [
            "Viewing ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "span",
            },
            " ",
            "expense",
            " totalling ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$2.35",
              },
              "ref": null,
              "rendered": "$2.35",
              "type": "span",
            },
            ".",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="button"
              replace={false}
              to="/create"
>
              Add Expense
</Link>,
            "className": "page-header__actions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add Expense",
              "className": "button",
              "replace": false,
              "to": "/create",
            },
            "ref": null,
            "rendered": "Add Expense",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content-container"
>
          <h1
                    className="page-header__title"
          >
                    Viewing 
                    <span>
                              1
                    </span>
                     
                    expense
                     totalling 
                    <span>
                              $2.35
                    </span>
                    .
          </h1>
          <div
                    className="page-header__actions"
          >
                    <Link
                              className="button"
                              replace={false}
                              to="/create"
                    >
                              Add Expense
                    </Link>
          </div>
</div>,
        "className": "page-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="page-header__title"
>
              Viewing 
              <span>
                            1
              </span>
               
              expense
               totalling 
              <span>
                            $2.35
              </span>
              .
</h1>,
            <div
              className="page-header__actions"
>
              <Link
                            className="button"
                            replace={false}
                            to="/create"
              >
                            Add Expense
              </Link>
</div>,
          ],
          "className": "content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Viewing ",
                <span>
                  1
</span>,
                " ",
                "expense",
                " totalling ",
                <span>
                  $2.35
</span>,
                ".",
              ],
              "className": "page-header__title",
            },
            "ref": null,
            "rendered": Array [
              "Viewing ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                },
                "ref": null,
                "rendered": 1,
                "type": "span",
              },
              " ",
              "expense",
              " totalling ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$2.35",
                },
                "ref": null,
                "rendered": "$2.35",
                "type": "span",
              },
              ".",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="button"
                replace={false}
                to="/create"
>
                Add Expense
</Link>,
              "className": "page-header__actions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Add Expense",
                "className": "button",
                "replace": false,
                "to": "/create",
              },
              "ref": null,
              "rendered": "Add Expense",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should render ExpensesSummary with 1+ expense 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={23}
    expensesTotal={234453}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content-container"
>
        <h1
                className="page-header__title"
        >
                Viewing 
                <span>
                        23
                </span>
                 
                expenses
                 totalling 
                <span>
                        $2,344.53
                </span>
                .
        </h1>
        <div
                className="page-header__actions"
        >
                <Link
                        className="button"
                        replace={false}
                        to="/create"
                >
                        Add Expense
                </Link>
        </div>
</div>,
      "className": "page-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-header__title"
>
            Viewing 
            <span>
                        23
            </span>
             
            expenses
             totalling 
            <span>
                        $2,344.53
            </span>
            .
</h1>,
          <div
            className="page-header__actions"
>
            <Link
                        className="button"
                        replace={false}
                        to="/create"
            >
                        Add Expense
            </Link>
</div>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Viewing ",
              <span>
                23
</span>,
              " ",
              "expenses",
              " totalling ",
              <span>
                $2,344.53
</span>,
              ".",
            ],
            "className": "page-header__title",
          },
          "ref": null,
          "rendered": Array [
            "Viewing ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 23,
              },
              "ref": null,
              "rendered": 23,
              "type": "span",
            },
            " ",
            "expenses",
            " totalling ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$2,344.53",
              },
              "ref": null,
              "rendered": "$2,344.53",
              "type": "span",
            },
            ".",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="button"
              replace={false}
              to="/create"
>
              Add Expense
</Link>,
            "className": "page-header__actions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add Expense",
              "className": "button",
              "replace": false,
              "to": "/create",
            },
            "ref": null,
            "rendered": "Add Expense",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content-container"
>
          <h1
                    className="page-header__title"
          >
                    Viewing 
                    <span>
                              23
                    </span>
                     
                    expenses
                     totalling 
                    <span>
                              $2,344.53
                    </span>
                    .
          </h1>
          <div
                    className="page-header__actions"
          >
                    <Link
                              className="button"
                              replace={false}
                              to="/create"
                    >
                              Add Expense
                    </Link>
          </div>
</div>,
        "className": "page-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="page-header__title"
>
              Viewing 
              <span>
                            23
              </span>
               
              expenses
               totalling 
              <span>
                            $2,344.53
              </span>
              .
</h1>,
            <div
              className="page-header__actions"
>
              <Link
                            className="button"
                            replace={false}
                            to="/create"
              >
                            Add Expense
              </Link>
</div>,
          ],
          "className": "content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Viewing ",
                <span>
                  23
</span>,
                " ",
                "expenses",
                " totalling ",
                <span>
                  $2,344.53
</span>,
                ".",
              ],
              "className": "page-header__title",
            },
            "ref": null,
            "rendered": Array [
              "Viewing ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 23,
                },
                "ref": null,
                "rendered": 23,
                "type": "span",
              },
              " ",
              "expenses",
              " totalling ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$2,344.53",
                },
                "ref": null,
                "rendered": "$2,344.53",
                "type": "span",
              },
              ".",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="button"
                replace={false}
                to="/create"
>
                Add Expense
</Link>,
              "className": "page-header__actions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Add Expense",
                "className": "button",
                "replace": false,
                "to": "/create",
              },
              "ref": null,
              "rendered": "Add Expense",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
